src/database.rs

```diff
@@ -89,6 +88,21 @@ pub trait Database: Send + AsDynDatabase + Any + ZalsaDatabase {
     {
         crate::attach::attach(self, || op(self))
     }
+
+    #[doc(hidden)]
+    fn zalsa_register_downcaster(&self) {
+        // The no-op downcaster is special cased in view caster construction.
+    }
+
+    #[doc(hidden)]
+    #[inline(always)]
+    unsafe fn downcast(db: &dyn Database) -> &dyn Database
+    where
+        Self: Sized,
+    {
+        // No-op
+        db
+    }
```

Comment:
Interestingly enough, this change also means that one does not actually need to use `#[salsa::db]` on trait impls for the baseline `Database`